File: projects/m1/015-is-it-a-leap-year/solutions/enricozoboli/solution_is_it_a_leap_year.py
user_year = int(input("Insert a year: "))


if user_year % 400 == 0 or (user_year % 100 != 0 and user_year % 4 == 0):
    year = "a leap"

else:
    year = "not a leap"

print(f"\n{user_year} is {year} year.\n")

File: projects/m1/015-is-it-a-leap-year/solutions/valentinaBazzacchi/solution-is-it-a-leap-year.js
let year = prompt("Insert a year!");
  
if (year % 400 == 0) {
alert("This is a leap year!");
} else if (year % 400 != 0 && year % 100 == 0) {
alert("This isn't a leap year!");
} else if (year % 100 != 0 && year % 4 == 0) {
alert("This is a leap year!");  
} else {
alert("This isn't a leap year!");  
}

// Versione più breve: inserisco tutte le condizioni in un'unica parentesi utilizzando gli operatori logici appropriati

let year = prompt("Insert a year!");
    
if ((year % 400 == 0) || year % 100 != 0 && year % 4 == 0) {
alert("This is a leap year!");
} else {
alert("This isn't a leap year!");    
}